### PR TITLE
Add setting to toggle interpreting square brackets as style

### PR DESCRIPTION
**Pull request by ahuang11**
Sometimes, strings that contain brackets  [  ]  aren't being logged correctly because rich interprets them as styles. This adds a setting to allow toggling and defaults it to False:

"Whether to interpret strings wrapped in square brackets as a style.
This allows styles to be conveniently added to log messages, e.g.
`[red]This is a red message.[/red]`. However, the downside is,
if enabled, strings that contain square brackets may be inaccurately
interpreted and lead to incomplete output, e.g.
`DROP TABLE [dbo].[SomeTable];"` results in `DROP TABLE .[SomeTable];`."

Originally from [here](https://prefect-community.slack.com/archives/CL09KU1K7/p1670455860648239)

### Example

```
from prefect import flow, task, get_run_logger

@task
def execute_sql(sql: str):
    logger = get_run_logger()
    logger.info(f"Executing: {sql}")
    logger.info(sql)


@flow
def bracket_flow():
    execute_sql("DROP TABLE [dbo].[SomeTable];")
    execute_sql("EXEC [dbo].[usp_SomeProcedure];")


if __name__ == "__main__":
    bracket_flow()
```

image

### Checklist

- [x] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`

### CODE REVIEW

**Review by tpdorsey — APPROVED** [review]
Docs changes look good to me. thx

File: src/prefect/settings.py [hunk 1]
```diff
@@ -562,6 +562,19 @@ def default_cloud_ui_url(settings, value):
 )
 """Whether to style console logs with color."""
 
+PREFECT_LOGGING_MARKUP = Setting(
```
Comment (madkinsz) [inline src/prefect/settings.py:565]: I think we often use `_ENABLED` for settings like this, but I see you didn't for the logging colors option so I don't feel strongly here.

**Review by madkinsz — APPROVED** [review]
Great work!

File: docs/concepts/logs.md [hunk 2]
```diff
@@ -333,6 +333,31 @@ def log_email_flow():
...
+PREFECT_LOGGING_MARKUP=True
+```
+</div>
+
+Then, the following will highlight "fancy" in red.
+```python
+from prefect import flow, get_run_logger
+
+@flow
+def log_email_flow():
```
Comment (ahuang11) [inline docs/concepts/logs.md:351]:
```suggestion
def my_flow():
```

**Pull request merged**